python/pyarrow/tests/test_types.py

```diff
@@ -1131,3 +1131,24 @@ def test_hashing(items):
 
     for i, item in enumerate(items):
         assert container[item] == i
+
+
+def test_types_come_back_with_specific_type():
+    for arrow_type in get_many_types():
+        schema = pa.schema([pa.field("field_name", arrow_type)])
+        type_back = schema.field("field_name").type
+        assert type(type_back) is type(arrow_type)
+
+
+@pytest.mark.parametrize("unit", ["us", "ns"])
+def test_can_read_unit_from_time_64_array(unit: str):
+    array = pa.array([1, 2, 3], pa.time64(unit))
+    assert array.type.unit == unit
+    assert isinstance(array.type, pa.Time64Type)
+
+
+@pytest.mark.parametrize("unit", ["s", "ms"])
+def test_can_read_unit_from_time_32_array(unit: str):
+    array = pa.array([1, 2, 3], pa.time32(unit))
+    assert array.type.unit == unit
+    assert isinstance(array.type, pa.Time32Type)
```

Comment:
I am not fully sure those two tests are needed. We already have `test_time32_units` and `test_time64_units` which cover the fact that those subclasses have a `unit` property. And the test you added above will ensure that this subclass is preserved when getting the type from an array of schema.

Reply:
I've removed these 2.